templates/project_stl_containers/MyProjGenerators/Pythia6Generator.cxx

```diff
@@ -28,113 +28,119 @@ using std::max;
 Pythia6Generator::Pythia6Generator() {}
 // ------------------------------------------------------------------------
 
-
-
 // -----   Standard constructor   -----------------------------------------
-Pythia6Generator::Pythia6Generator(const char* fileName) {
-  fFileName  = fileName;
-  fVerbose = 0;
-  cout << "-I Pythia6Generator: Opening input file " << fileName << endl;
-  if ((fInputFile = fopen(fFileName,"r"))==NULL)
-    //  fInputFile = new ifstream(fFileName);
-    //  if ( ! fInputFile->is_open() ) 
-    LOG(fatal) << "Pythia6Generator: Cannot open input file.";
-  
-  // fPDG=TDatabasePDG::Instance();
+Pythia6Generator::Pythia6Generator(const char* fileName)
+{
+    fFileName = fileName;
+    fVerbose = 0;
+    cout << "-I Pythia6Generator: Opening input file " << fileName << endl;
+    if ((fInputFile = fopen(fFileName, "r")) == NULL)
+        //  fInputFile = new ifstream(fFileName);
+        //  if ( ! fInputFile->is_open() )
+        LOG(fatal) << "Pythia6Generator: Cannot open input file.";
+
+    // fPDG=TDatabasePDG::Instance();
 }
 // ------------------------------------------------------------------------
 
-
-
 // -----   Destructor   ---------------------------------------------------
-Pythia6Generator::~Pythia6Generator() {
-  CloseInput();
-}
+Pythia6Generator::~Pythia6Generator() { CloseInput(); }
 // ------------------------------------------------------------------------
 
+// -----   Public method ReadEvent   --------------------------------------
+Bool_t Pythia6Generator::ReadEvent(FairPrimaryGenerator* primGen)
+{
+
+    // Check for input file
+    if (!fInputFile) {
+        // if ( ! fInputFile->is_open() ) {
+        cout << "-E Pythia6Generator: Input file not open!" << endl;
+        return kFALSE;
+    }
+
+    // Define event variable to be read from file
+    Int_t ntracks = 0, eventID = 0, ncols = 0;
+
+    // Define track variables to be read from file
+    Int_t nLev = 0, pdgID = 0, nM1 = -1, nM2 = -1, nDF = -1, nDL = -1;
+    Float_t fPx = 0., fPy = 0., fPz = 0., fM = 0., fE = 0.;
+    Float_t fVx = 0., fVy = 0., fVz = 0., fT = 0.;
+
+    // Read event header line from input file
+
+    Int_t max_nr = 0;
+
+    Text_t buffer[200];
+    ncols = fscanf(fInputFile, "%d\t%d", &eventID, &ntracks);
+
+    if (ncols && ntracks > 0) {
+
+        if (fVerbose > 0)
+            cout << "Event number: " << eventID << "\tNtracks: " << ntracks << endl;
+
+        for (Int_t ll = 0; ll < ntracks; ll++) {
+            ncols = fscanf(fInputFile,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'ncols' is assigned a value that is never used.](https://app.codacy.com/manual/FairRootGroup/FairRoot/pullRequest?prid=5323596)